patterns/design-patterns-architectural/src/test/java/com/baeldung/seda/springintegration/SpringIntegrationSedaUnitTest.java

```diff
@@ -0,0 +1,54 @@
+package com.baeldung.seda.springintegration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.integration.annotation.IntegrationComponentScan;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import com.baeldung.seda.springintegration.ChannelConfiguration;
+import com.baeldung.seda.springintegration.IntegrationConfiguration;
+import com.baeldung.seda.springintegration.TaskExecutorConfiguration;
+import com.baeldung.seda.springintegration.TestGateway;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(loader = AnnotationConfigContextLoader.class, classes = { TaskExecutorConfiguration.class, ChannelConfiguration.class, IntegrationConfiguration.class })
+@EnableIntegration
+@IntegrationComponentScan(basePackages = { "com.baeldung.seda.springintegration" })
+public class SpringIntegrationSedaUnitTest {
```

Comment:
Per the "for authors - 4 - implementation code (new)" doc, the Spring config means this should run as part of the integration suite. i.e. the suffix should be "IntegrationTest" instead of "UnitTest"

Reply:
I think this would apply to CamelTestSupport, too, depending on how much -- if anything -- that class is mocking.